base/server/src/main/java/com/netscape/cmscore/dbs/LDAPSession.java

```diff
@@ -474,6 +476,61 @@ public DBSearchResults search(String base, String filter,
         }
     }
 
+    @Override
+    public DBSearchResults pagedSearch(String base, String filter, int start, int size, int timeLimit)
+            throws EBaseException {
+        try {
+            String ldapfilter = dbSubsystem.getRegistry().getFilter(filter);
+            logger.info("LDAPSession.pagedSearch(): Searching {}  for {}", base, ldapfilter);
+
+            LDAPSearchConstraints cons = new LDAPSearchConstraints();
+            if (timeLimit > 0) {
+                cons.setServerTimeLimit(timeLimit);
```

Comment:
The `LDAPSearchConstraints` also has a `setMaxResults()`. I was wondering whether we should just ignore it, set it to `size`, or add another param for it (e.g. `sizeLimit`).

Reply:
Another consideration is, IIUC the `size` is only used internally for paging, but the number of results the client will see won't actually be affected by the `size`. On the other hand, the `sizeLimit` will (if it's used to call `setMaxResults()`).

So maybe we should replace `size` with a constant or a server-side configurable param, and add a `sizeLimit` not for paging but for setting the max results.

Reply:
Actually,  `size` elements are returned from the search. The size is always used in the last query and the elements are returned . If start is provided, than one or more pages are requested until `start - 1` elements are retrieved and then a page with `size` is requested.
If `size` is not provided there is a limit to 500. I am leaving the limit now and we can decide if this has to be customisable in a separate PR